lib/vre/ch31_form.rb

```diff
@@ -63,49 +63,71 @@ def format_payload_for_vre
       }
 
       vre_payload[:data].merge!(veteran_address(form_data))
-      vre_payload[:data].merge!({ veteranInformation: claim_form_hash['veteranInformation'] })
+      vre_payload[:data].merge!({ veteranInformation: adjusted_veteran_information })
       vre_payload[:data].merge!(new_address) if claim_form_hash['newAddress'].present?
 
       vre_payload.to_json
     end
 
     def veteran_address(form_data)
-      vet_address = form_data['veteranAddress']
+      vet_address = mapped_address_hash(form_data['veteranAddress'])
 
-      {
-        veteranAddress: {
-          isForeign: vet_address['country'] != 'USA',
-          isMilitary: vet_address['isMilitary'] || false,
-          countryName: vet_address['country'],
-          addressLine1: vet_address['street'],
-          addressLine2: vet_address['street2'],
-          addressLine3: vet_address['street3'],
-          city: vet_address['city'],
-          stateCode: vet_address['state'],
-          zipCode: vet_address['postalCode']
-        }
+      adjusted_address = {
+        veteranAddress: vet_address
       }
+
+      return adjusted_address if adjusted_address.dig(:veteranAddress, :isForeign) == false
+
+      international_address = adjusted_address[:veteranAddress]
+      international_address[:internationPostalCode] = international_address.delete(:zipCode)
+      international_address[:province] = international_address.delete(:stateCode)
+
+      adjusted_address
     end
 
     def claim_form_hash
       @claim.parsed_form
     end
 
+    def adjusted_veteran_information
+      vet_info = claim_form_hash['veteranInformation'].with_indifferent_access
+
+      vet_info['VAFileNumber'] = vet_info.delete('vaFileNumber') if vet_info.key?('vaFileNumber')
+
+      vet_info['VAFileNumber'] = vet_info['ssn'] if vet_info['VAFileNumber'].nil?
+
+      vet_info
+    end
+
     def new_address
-      new_address = claim_form_hash['newAddress']
-      {
-        "newAddress": {
-          "isForeign": new_address['country'] != 'USA',
-          "isMilitary": new_address['isMilitary'],
-          "countryName": new_address['country'],
-          "addressLine1": new_address['street'],
-          "addressLine2": new_address['street2'],
-          "addressLine3": new_address['street3'],
-          "city": new_address['city'],
-          "province": new_address['state'],
-          "internationalPostalCode": new_address['postalCode']
-        }
+      new_address = mapped_address_hash(claim_form_hash['newAddress'])
+
+      adjusted_new_address = {
+        newAddress: new_address
       }
+
+      return adjusted_new_address if adjusted_new_address.dig(:newAddress, :isForeign) == false
+
+      new_vet_international_address = adjusted_new_address[:newAddress]
```

Comment:
I would just use `newAddress` here instead of declaring a new variable.  If you're worried about the context, this would be a good place for a comment like `# change zipCode and stateCode to the international names`